src/toil/cwl/cwltoil.py

```diff
@@ -2345,6 +3044,7 @@ def set_secondary(fileobj):
             # make sure this doesn't add listing items; if shallow_listing is
             # selected, it will discover dirs one deep and then again later on
             # (producing 2+ deep listings instead of only 1)
+            # TODO: when is later on?
```

Comment:
When the builder operates on this in the `job.run()` step if I remember correctly.  Though I may be misremembering.